src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3813,23 +3813,40 @@ bool shouldConvertToIDisposableBeforeTry(IOperation resource)
 
             void processResource(IOperation resource, ArrayBuilder<(IVariableDeclarationOperation, IVariableDeclaratorOperation)>? resourceQueueOpt)
             {
-                // When ResourceType is a non-nullable value type, the expansion is:
+                // When ResourceType is a non-nullable value type that implements IDisposable, the expansion is:
                 //
                 // {
                 //   ResourceType resource = expr;
                 //   try { statement; }
                 //   finally { ((IDisposable)resource).Dispose(); }
                 // }
                 //
+                // Otherwise, when ResourceType is a non-nullable value type that implements
+                // disposal via pattern, the expansion is:
+                //
+                // {
+                //   try { statement; }
+                //   finally { resource.Dispose(); }
+                // }
                 // Otherwise, when Resource type is a nullable value type or
-                // a reference type other than dynamic, the expansion is:
+                // a reference type other than dynamic that implements IDisposable, the expansion is:
                 //
                 // {
                 //   ResourceType resource = expr;
                 //   try { statement; }
                 //   finally { if (resource != null) ((IDisposable)resource).Dispose(); }
                 // }
                 //
+                // Otherwise, when Resource type is a nullable value type or
+                // a reference type other than dynamic that implements disposal
+                // via pattern, the expansion is:
+                //
+                // {
+                //   ResourceType resource = expr;
+                //   try { statement; }
+                //   finally { if (resource != null) resource.Dispose(); }
```

Comment:
>resource [](start = 53, length = 8)

Is this accurate for a nullable value type? We are not unwrapping the ```Nullable<T>```?

Reply:
Updated the comment to show what we do in lowering.

Reply:
> Updated the comment to show what we do in lowering.

Why would you do this? Is this comment for lowering? Is this comment for lowering of a specific compiler?

---
In reply to: [534592569](https://github.com/dotnet/roslyn/pull/49481#discussion_r534592569) [](ancestors = 534592569)

Reply:
I understood your comment to mean that we should be showing the unwrapping of the nullable in CFG, but clearly I misunderstood you.

Reply:
> I understood your comment to mean that we should be showing the unwrapping of the nullable in CFG, but clearly I misunderstood you.

I asked you about behavior that you are describing in a new comment specifically for a pattern-based case. I couldn't find a test for it, hence the question. Clearly the comment looked strange to me, but you started looking at completely different scenario.

---
In reply to: [534620372](https://github.com/dotnet/roslyn/pull/49481#discussion_r534620372) [](ancestors = 534620372)

Reply:
Apologies, as I said, I misunderstood what you were asking.